src/clowder.ts

```diff
@@ -310,7 +312,6 @@ export class Clowder {
         return unvalidatedRemainingItems[randInt];
```

Comment:
We should handle the `returnUndefinedOnExhaustion = false` case here as well. If the user has `catToSelect = 'unvalidated'` and `returnUndefinedOnExhaustion = false`, then when `unvalidatedRemainingItems.length === 0`, instead of returning `undefined`, we should return an item that does have parameters for some cat.

Perhaps we can move the
```
const { available, missing } = filterItemsByCatParameterAvailability(this._remainingItems, catToSelect);
```

line above the `if (catToSelect === 'unvalidated') {` line and then draw an item from `missing` in this case.

Reply:
done!